cpp/src/jni/dataset/proto/record_batch.proto

```diff
@@ -0,0 +1,55 @@
+// Licensed to the Apache Software Foundation (ASF) under one
```

Comment:
why are we using protos for this information?

Reply:
In particular this for the most part looks like it replicates data already defined in flatbuffers.

Reply:
Migrated to flatbuffers in PR #10201